src/Bridge/Doctrine/MongoDB/CollectionDataProvider.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDB;
+
+use ApiPlatform\Core\Api\CollectionDataProviderInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ODM\MongoDB\DocumentRepository;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryResultExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryCollectionExtensionInterface;
+
+/**
+ * Collection data provider for the Doctrine ORM.
+ */
+class CollectionDataProvider implements CollectionDataProviderInterface
```

Comment:
Is this exactly the same class that the class for the ORM?

If it is the case, this class can be moved in the `Doctrine` and used by both instead of being duplicated. If changes are small, an `abstract` class can be introduced.